src/replica_server/replication_lib/mutation.cpp

```diff
@@ -285,7 +286,18 @@ mutation_queue::mutation_queue(gpid gpid, int max_concurrent_op /*= 2*/, bool ba
 
 mutation_ptr mutation_queue::add_work(task_code code, dsn_message_t request, replica* r)
 {
-    // batch and add to work queue
+    task_spec* spec = task_spec::get(code);
+
+    // if not allow write batch, switch work queue
+    if (_pending_mutation && !spec->rpc_request_is_write_allow_batch)
```

Comment:
it is a time-consuming work to fix it, or we can leave a issue to the future, not blocking the open source work of pegasus?